### PR TITLE
Fixed Notebook Tests

### LINKED ISSUE #41: Tests jupyter notebooks
Currently all the source code we have is in Jupyter notebooks and we want to make sure they run so we are testing it with nbmake.

### PR DESCRIPTION
**Pull request by pjflux2001**
* Added a flag file to differentiate between workflow tests
* Made changes in notebook dir path and few minor changes in notebook itself
* Added few requirements which were required for staging notebook
* Upgraded macOS versions as previous ones are deprecated [read issue here](https://github.com/actions/runner-images/issues/5583)

Closes #41

### CODE REVIEW

File: .github/workflows/ubuntu20_python.yml [hunk 1]
```diff
@@ -30,6 +30,9 @@ jobs:
         python -m pip install flake8 pytest
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
         pip install . -U
+    - name: Create Flag File
+      run: |
+        touch /tmp/flag
```
Comment (PatriceJada) [inline .github/workflows/ubuntu20_python.yml:35]: I have not used this but do we really need it?

File: src/polyphy/profile.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (PatriceJada) [inline src/polyphy/profile.json:1]: Do we really need this file?

**Review by PatriceJada — APPROVED** [review]
@pjflux2001 thank you for getting this working. Looks good but I just put up a few concerns. Do we need the production folder? `PolyPhy/experiments/jupyter
/production/` The path looks very long. Could you also squash your commits to one issue. The conventional way should be one commit per issue fixed, this avoids us getting passed the 2GB that git provides.

File: .github/workflows/ubuntu20_python.yml (hunk 1 above)
Comment (pjflux2001) [reply .github/workflows/ubuntu20_python.yml:35]: Yes, this is needed. Might not be most efficient way to do it - but I couldn't think of a better approach yet

File: src/polyphy/profile.json (hunk 2 above)
Comment (pjflux2001) [reply src/polyphy/profile.json:1]: Removing it

**Comment by pjflux2001** [comment]
> @pjflux2001 thank you for getting this working. Looks good but I just put up a few concerns. Do we need the production folder? `PolyPhy/experiments/jupyter /production/` The path looks very long. Could you also squash your commits to one issue. The conventional way should be one commit per issue fixed, this avoids us getting passed the 2GB that git provides.

Since the 3D mask jupyter notebook isn't yet working / incomplete, I checked with @OskarElek and he suggested segregating it from the other notebooks. Hence, this change in path. 

@OskarElek could you please turn on squashing : https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

**Pull request merged**